Mathlib/Order/LiminfLimsup.lean

```diff
@@ -1517,3 +1517,98 @@ theorem liminf_finset_inf [ConditionallyCompleteLinearOrder β] [OrderTop β] {f
   limsup_finset_sup (β := βᵒᵈ) h₁ h₂
 
 end MinMax
+
+section frequently_bounded
+
+variable {R S : Type*} {F : Filter R} [LinearOrder R] [LinearOrder S]
+
+/-- For nontrivial filters in linear orders, coboundedness for `≤` implies frequently boundedness
+from below. -/
+lemma IsCobounded.frequently_ge [NeBot F] (cobdd : IsCobounded (· ≤ ·) F) :
+    ∃ l, ∃ᶠ x in F, l ≤ x := by
+  obtain ⟨t, ht⟩ := cobdd
+  by_cases tbot : IsBot t
+  · refine ⟨t, frequently_of_forall fun r ↦ tbot r⟩
+  obtain ⟨t', ht'⟩ : ∃ t', t' < t := by
+    by_contra!
+    exact tbot this
+  refine ⟨t', ?_⟩
+  intro ev
+  specialize ht t' (by filter_upwards [ev] with _ h using (not_le.mp h).le)
+  apply lt_irrefl t' <| lt_of_lt_of_le ht' ht
+
+/-- For nontrivial filters in linear orders, coboundedness for `≥` implies frequently boundedness
+from above. -/
+lemma IsCobounded.frequently_le [NeBot F] (cobdd : IsCobounded (· ≥ ·) F) :
+    ∃ u, ∃ᶠ x in F, x ≤ u :=
+  IsCobounded.frequently_ge (R := Rᵒᵈ) cobdd
+
+/-- In linear orders, frequently boundedness from below implies coboundedness for `≤`. -/
+lemma isCobounded_le_of_frequently_ge {l : R} (freq_ge : ∃ᶠ x in F, l ≤ x) :
```

Comment:
We're now putting stuff like this in the `IsCobounded` namespace. It allows for dot notation (where there is no text preceding the dot and the expected type is known).

Reply:
Thanks! This made me realize that I had made a mistake in the namespaces, because they need to be further under the `Filter` namespace (so `Filter.IsCobounded` etc.). I did not right away get the dot notation working, however, but that is probably my ignorance of the syntax and from my point of view the naming in `Filter.IsCobounded` namespace would be ok in any case. I changed the naming.